pkg/infra/tracing/opentelemetry_tracing.go

```diff
@@ -86,25 +118,82 @@ func (ots *Opentelemetry) initTracerProvider() (*tracesdk.TracerProvider, error)
 	return tp, nil
 }
 
-func (ots *Opentelemetry) initOpentelemetryTracer() error {
-	tp, err := ots.initTracerProvider()
+func (ots *Opentelemetry) initOTLPTracerProvider() (*tracesdk.TracerProvider, error) {
+	client := otlptracegrpc.NewClient(otlptracegrpc.WithEndpoint(ots.address), otlptracegrpc.WithInsecure())
+	exp, err := otlptrace.New(context.Background(), client)
 	if err != nil {
-		return err
+		return nil, err
+	}
+
+	res, err := resource.New(
+		context.Background(),
+		resource.WithAttributes(
+			semconv.ServiceNameKey.String("grafana"),
+			semconv.ServiceVersionKey.String(version.Version),
+		),
+		resource.WithProcessRuntimeDescription(),
+		resource.WithTelemetrySDK(),
+	)
+	if err != nil {
+		return nil, err
 	}
+
+	tp := tracesdk.NewTracerProvider(
+		tracesdk.WithBatcher(exp),
+		tracesdk.WithSampler(tracesdk.ParentBased(
+			tracesdk.AlwaysSample(),
```

Comment:
I'm not too familiar with how Grafana uses tracing but is there a switch somewhere to turn tracing off?